bin/aws-configure

```diff
@@ -39,72 +31,55 @@ To add a new profile, you MUST specify following environment variables:
 
 You can optionally specify '<PROFILE>_AWS_DEFAULT_REGION' to determine the default region for this profile.
 
-If no PROFILE is specified, ${0##*/} will defer to the value specified in AWS_PROFILE.
-
 SUBCOMMANDS:
     is-configured PROFILE     Returns '0' if the profile is configured. Otherwise, it returns '1'.
 DOC
 }
 
 function setup_profile() {
-  profile="$1"
-  profile_env="$(echo "${profile^^}" | tr '-' '_')"
+  profile=$1
+  profile_env="$(echo "$profile" | awk '{ print toupper($0) }' | tr '-' '_')"
 
-  enforce_requirements
-
-  if [[ -z "$profile" ]];
-  then
-    echo -e "${0##*/}: No profile specified. You must specify an AWS profile to configure.\n"
+  if [[ -z "$profile" ]]; then
+    echo -e "${0##*/}: No profile specified. You must specify an AWS profile to configure.\\n"
     usage
     exit 2
   fi
 
-  [[ ! -d $HOME/.aws ]] && mkdir -p $HOME/.aws > /dev/null
+  [[ ! -d "$HOME/.aws" ]] && mkdir -p "$HOME/.aws" > /dev/null
 
   access_key_id="${profile_env}_AWS_ACCESS_KEY_ID"
   secret_access_key="${profile_env}_AWS_SECRET_ACCESS_KEY"
   region="${profile_env}_AWS_DEFAULT_REGION"
 
-  if [[ -z "${!access_key_id}" || -z "${!secret_access_key}" ]];
-  then
+  if [[ -z "${!access_key_id}" || -z "${!secret_access_key}" ]]; then
     echo "${0##*/}: Unable to find the required environment variables: $access_key_id and $secret_access_key."
     echo "Make sure that both are specified."
     exit 2
   fi
 
-  if is_configured $1
-  then
+  set +e
+  if ( is_configured "$profile" ); then
```

Comment:
There's no need to `set +e`. Even if `set -e` is in effect, the script won't exit if there's a failure in a conditional context. That is:

```bash
set -e
if false; then
  : # no-op
else
  echo "got here"
fi
false || echo "this kind of conditional also won't exit"
false # exits here
echo "didn't get here"
```

So, I think you can make this
```bash
if is_configured "$profile"; then
```
I don't see a need for a subshell in this instance. However, it can be useful if you want to temporarily `set +e` and automatically reset it:
```bash
set -e
# commands that you want to exit on failure
( set +e
  # commands that you don't want to exit on failure
  false
  echo "didn't exit"
)
# set -e is in effect again
false
echo "won't get here"
```
Subshells can also be useful to "try" a sequence of commands that you may abort, but don't want to exit the outer script:
```bash
set -e
(
  echo "doing things that may fail"
  false
) || echo "subshell returned failure"
echo "didn't exit whole script"
```

Reply:
Oh, I see why you're doing the subshell, because `is_configured` calls `exit`. I'd suggest changing that to
```bash
function is_configured() {
  aws --profile "$1" configure list
}
```
as long as you call `is_configured` from a conditional context like you are already doing, a failing command within the function won't exit the script.